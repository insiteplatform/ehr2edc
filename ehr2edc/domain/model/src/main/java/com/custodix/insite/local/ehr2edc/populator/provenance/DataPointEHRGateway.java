package com.custodix.insite.local.ehr2edc.populator.provenance;

import java.util.List;

import com.custodix.insite.local.ehr2edc.vocabulary.SubjectId;

public interface DataPointEHRGateway {

	List<String> findAllForSubject(SubjectId subjectId);

}
