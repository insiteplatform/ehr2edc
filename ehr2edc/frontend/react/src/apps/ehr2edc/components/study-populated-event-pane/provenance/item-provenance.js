import React from "react";
import RequestError from "../../../../../common/components/error/backend-request-error";

export class ItemProvenance extends React.Component {

    render() {
        const {item} = this.props;
        return !!item ? <>
            <h4 className="facts-title">Source Data Details</h4>
            {this.info(item)}
            {this.provenance(item)}
        </> : null;
    }

    info({name, value}) {
        return <dl className="margin-none">
            <h5>Name</h5>
            <dd>{name}</dd>
            <h5>Submitted Value</h5>
            <dd>{value}</dd>
        </dl>;
    }

    provenance({provenance}) {
        if (!!provenance) {
            const {error, data} = provenance;
            return <>
                <h5>Source Data</h5>
                {!error && !data ? this.provenanceLoading() : this.provenanceLoaded(provenance)}
            </>;
        }
    }

    provenanceLoading() {
        return <p>Loading...</p>
    }

    provenanceLoaded(provenance) {
        const {error, data} = provenance;
        return <>
            {this.provenanceError(error)}
            {this.provenanceDataListing(data)}
        </>;
    }

    provenanceError(error) {
        return !!error ?
            <div className="alert-box alert warning margin-trl">
                <RequestError error={error}/>
            </div> : null;
    }

    provenanceDataListing(data) {
        return !!data ? <div>
            {this.renderGroups(data.groups)}
            {this.renderItems(data.items)}
        </div> : null;
    }

    renderGroups(groups) {
        return groups.map((group, idx) => {
            const {label, items} = group;
            return <div key={idx} className="row">
                <h6>{label}</h6>
                {this.renderItems(items)}
            </div>
        })
    }

    renderItems(items) {
        return <dl className="responsive">
            {items.map((item, idx) => {
                const {label, value} = item;
                return <div key={idx} className="row">
                    <dt>{label}</dt>
                    <dd>{this.renderItemValue(value)}</dd>
                </div>;
            })}
        </dl>
    }

    renderItemValue(value) {
        return !!value ? value : <i className="disabled">none</i>
    }
}