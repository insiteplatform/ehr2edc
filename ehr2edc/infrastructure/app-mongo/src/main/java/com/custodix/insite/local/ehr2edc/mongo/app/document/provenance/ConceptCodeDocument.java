package com.custodix.insite.local.ehr2edc.mongo.app.document.provenance;

import org.springframework.data.annotation.PersistenceConstructor;

import com.custodix.insite.local.ehr2edc.provenance.model.ConceptCode;

final class ConceptCodeDocument {
	private final String code;

	@PersistenceConstructor
	private ConceptCodeDocument(String code) {
		this.code = code;
	}

	public static ConceptCodeDocument toDocument(ConceptCode concept) {
		return new ConceptCodeDocument(concept.getCode());
	}

	public ConceptCode restore() {
		return ConceptCode.of(code);
	}
}
