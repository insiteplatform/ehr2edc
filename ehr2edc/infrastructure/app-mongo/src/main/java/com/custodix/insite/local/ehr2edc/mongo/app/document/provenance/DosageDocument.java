package com.custodix.insite.local.ehr2edc.mongo.app.document.provenance;

import java.math.BigDecimal;

import org.springframework.data.annotation.PersistenceConstructor;

import com.custodix.insite.local.ehr2edc.provenance.model.Dosage;

final class DosageDocument {
	private final BigDecimal value;
	private final String unit;

	@PersistenceConstructor
	private DosageDocument(BigDecimal value, String unit) {
		this.value = value;
		this.unit = unit;
	}

	private DosageDocument(Builder builder) {
		value = builder.value;
		unit = builder.unit;
	}

	public static DosageDocument toDocument(Dosage dosage) {
		return DosageDocument.newBuilder()
				.withValue(dosage.getValue())
				.withUnit(dosage.getUnit())
				.build();
	}

	public Dosage restore() {
		return Dosage.newBuilder()
				.withValue(value)
				.withUnit(unit)
				.build();
	}

	public static Builder newBuilder() {
		return new Builder();
	}

	public static final class Builder {
		private BigDecimal value;
		private String unit;

		private Builder() {
		}

		public Builder withValue(BigDecimal value) {
			this.value = value;
			return this;
		}

		public Builder withUnit(String unit) {
			this.unit = unit;
			return this;
		}

		public DosageDocument build() {
			return new DosageDocument(this);
		}
	}
}
