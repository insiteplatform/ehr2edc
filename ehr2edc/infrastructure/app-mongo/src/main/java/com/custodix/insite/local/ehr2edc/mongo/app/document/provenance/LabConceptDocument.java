package com.custodix.insite.local.ehr2edc.mongo.app.document.provenance;

import java.util.Optional;

import org.springframework.data.annotation.PersistenceConstructor;

import com.custodix.insite.local.ehr2edc.provenance.model.FastingStatus;
import com.custodix.insite.local.ehr2edc.provenance.model.LabConcept;
import com.custodix.insite.local.ehr2edc.provenance.model.ConceptCode;

final class LabConceptDocument {
	private final ConceptCodeDocument concept;
	private final String component;
	private final String method;
	private final FastingStatus fastingStatus;
	private final String specimen;

	@PersistenceConstructor
	private LabConceptDocument(ConceptCodeDocument concept, String component, String method,
			FastingStatus fastingStatus, String specimen) {
		this.concept = concept;
		this.component = component;
		this.method = method;
		this.fastingStatus = fastingStatus;
		this.specimen = specimen;
	}

	private LabConceptDocument(Builder builder) {
		concept = builder.concept;
		component = builder.component;
		method = builder.method;
		fastingStatus = builder.fastingStatus;
		specimen = builder.specimen;
	}

	public static LabConceptDocument toDocument(LabConcept labConcept) {
		ConceptCodeDocument concept = Optional.ofNullable(labConcept.getConcept())
				.map(ConceptCodeDocument::toDocument)
				.orElse(null);
		return LabConceptDocument.newBuilder()
				.withConcept(concept)
				.withComponent(labConcept.getComponent())
				.withMethod(labConcept.getMethod())
				.withFastingStatus(labConcept.getFastingStatus())
				.withSpecimen(labConcept.getSpecimen())
				.build();
	}

	public LabConcept restore() {
		ConceptCode conceptCode = Optional.ofNullable(concept)
				.map(ConceptCodeDocument::restore)
				.orElse(null);
		return LabConcept.newBuilder()
				.withConcept(conceptCode)
				.withComponent(component)
				.withMethod(method)
				.withFastingStatus(fastingStatus)
				.withSpecimen(specimen)
				.build();
	}

	public static Builder newBuilder() {
		return new Builder();
	}

	public static final class Builder {
		private ConceptCodeDocument concept;
		private String component;
		private String method;
		private FastingStatus fastingStatus;
		private String specimen;

		private Builder() {
		}

		public Builder withConcept(ConceptCodeDocument concept) {
			this.concept = concept;
			return this;
		}

		public Builder withComponent(String component) {
			this.component = component;
			return this;
		}

		public Builder withMethod(String method) {
			this.method = method;
			return this;
		}

		public Builder withFastingStatus(FastingStatus fastingStatus) {
			this.fastingStatus = fastingStatus;
			return this;
		}

		public Builder withSpecimen(String specimen) {
			this.specimen = specimen;
			return this;
		}

		public LabConceptDocument build() {
			return new LabConceptDocument(this);
		}
	}
}
