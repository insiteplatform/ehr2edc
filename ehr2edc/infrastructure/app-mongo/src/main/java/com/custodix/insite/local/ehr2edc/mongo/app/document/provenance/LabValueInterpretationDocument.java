package com.custodix.insite.local.ehr2edc.mongo.app.document.provenance;

import org.springframework.data.annotation.PersistenceConstructor;

import com.custodix.insite.local.ehr2edc.provenance.model.LabValueInterpretation;

final class LabValueInterpretationDocument {
	private final Integer parsedInterpretation;
	private final String originalInterpretation;

	@PersistenceConstructor
	private LabValueInterpretationDocument(Integer parsedInterpretation, String originalInterpretation) {
		this.parsedInterpretation = parsedInterpretation;
		this.originalInterpretation = originalInterpretation;
	}

	private LabValueInterpretationDocument(Builder builder) {
		parsedInterpretation = builder.parsedInterpretation;
		originalInterpretation = builder.originalInterpretation;
	}

	public static LabValueInterpretationDocument toDocument(LabValueInterpretation qualitativeResult) {
		return LabValueInterpretationDocument.newBuilder()
				.withParsedInterpretation(qualitativeResult.getParsedInterpretation())
				.withOriginalInterpretation(qualitativeResult.getOriginalInterpretation())
				.build();
	}

	public LabValueInterpretation restore() {
		return LabValueInterpretation.newBuilder()
				.withParsedInterpretation(parsedInterpretation)
				.withOriginalInterpretation(originalInterpretation)
				.build();
	}

	public static Builder newBuilder() {
		return new Builder();
	}

	public static final class Builder {
		private Integer parsedInterpretation;
		private String originalInterpretation;

		private Builder() {
		}

		public Builder withParsedInterpretation(Integer parsedInterpretation) {
			this.parsedInterpretation = parsedInterpretation;
			return this;
		}

		public Builder withOriginalInterpretation(String originalInterpretation) {
			this.originalInterpretation = originalInterpretation;
			return this;
		}

		public LabValueInterpretationDocument build() {
			return new LabValueInterpretationDocument(this);
		}
	}
}
