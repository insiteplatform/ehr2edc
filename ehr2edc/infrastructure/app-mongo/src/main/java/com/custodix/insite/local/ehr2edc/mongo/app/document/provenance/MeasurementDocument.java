package com.custodix.insite.local.ehr2edc.mongo.app.document.provenance;

import java.math.BigDecimal;

import org.springframework.data.annotation.PersistenceConstructor;

import com.custodix.insite.local.ehr2edc.provenance.model.Measurement;

final class MeasurementDocument {
	private final BigDecimal lowerLimit;
	private final BigDecimal upperLimit;
	private final BigDecimal value;
	private final String unit;

	@PersistenceConstructor
	private MeasurementDocument(BigDecimal lowerLimit, BigDecimal upperLimit, BigDecimal value, String unit) {
		this.lowerLimit = lowerLimit;
		this.upperLimit = upperLimit;
		this.value = value;
		this.unit = unit;
	}

	private MeasurementDocument(Builder builder) {
		lowerLimit = builder.lowerLimit;
		upperLimit = builder.upperLimit;
		value = builder.value;
		unit = builder.unit;
	}

	public static MeasurementDocument toDocument(Measurement quantitativeResult) {
		return MeasurementDocument.newBuilder()
				.withLowerLimit(quantitativeResult.getLowerLimit())
				.withUpperLimit(quantitativeResult.getUpperLimit())
				.withValue(quantitativeResult.getValue())
				.withUnit(quantitativeResult.getUnit())
				.build();
	}

	public Measurement restore() {
		return Measurement.newBuilder()
				.withLowerLimit(lowerLimit)
				.withUpperLimit(upperLimit)
				.withValue(value)
				.withUnit(unit)
				.build();
	}

	public static Builder newBuilder() {
		return new Builder();
	}

	public static final class Builder {
		private BigDecimal lowerLimit;
		private BigDecimal upperLimit;
		private BigDecimal value;
		private String unit;

		private Builder() {
		}

		public Builder withLowerLimit(BigDecimal lowerLimit) {
			this.lowerLimit = lowerLimit;
			return this;
		}

		public Builder withUpperLimit(BigDecimal upperLimit) {
			this.upperLimit = upperLimit;
			return this;
		}

		public Builder withValue(BigDecimal value) {
			this.value = value;
			return this;
		}

		public Builder withUnit(String unit) {
			this.unit = unit;
			return this;
		}

		public MeasurementDocument build() {
			return new MeasurementDocument(this);
		}
	}
}
