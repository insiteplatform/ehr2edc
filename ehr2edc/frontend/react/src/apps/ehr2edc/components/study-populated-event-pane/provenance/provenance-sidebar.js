import React from "react"
import {Sidebar} from "../../../../../common/components/sidebar/sidebar"
import {Portal} from "react-overlays";
import {ItemProvenance} from "./item-provenance";
import {connect} from "react-redux";
import {
    closeItemDetails,
    fetchItemProvenance,
    fetchSubmittedItemProvenance
} from "../../../actions/item-provenance.actions";

const SidebarPortal = ({children}) => {
    const container = document.getElementById("study-details-content-pane");
    return <Portal container={container}>
        {children}
    </Portal>
};

export class ProvenanceSidebar extends React.Component {

    constructor(props) {
        super(props);
        this.state = {};

        this.onClose = this.onClose.bind(this);
    }

    componentWillUnmount() {
        if (this.hasData()) {
            this.onClose();
        }
    }

    componentDidUpdate(prevProps) {
        if (this.shouldFetchProvenance(prevProps)) {
            this.fetchProvenance();
        }
    }

    shouldFetchProvenance(previous) {
        const {eventId: prevEventId, subjectId: prevSubjectId, itemId: prevItemId} = previous;
        const {eventId, subjectId, itemId} = this.props;
        return !!eventId && eventId !== prevEventId
            || !!subjectId && subjectId !== prevSubjectId
            || !!itemId && itemId !== prevItemId;
    }

    fetchProvenance() {
        const {actions, studyId, eventId, subjectId, itemId, inReview} = this.props;
        if (!!inReview) {
            actions.fetchItemProvenance({studyId, eventId, subjectId, itemId});
        } else {
            actions.fetchSubmittedItemProvenance({studyId, eventId, subjectId, itemId})
        }
    }

    render() {
        const {name, value, provenance} = this.props;
        return this.hasData()
            ? <SidebarPortal>
                <Sidebar additionalClasses={["small-4", "border-left", "slideInLeft", "slideOutRight"]}>
                    <div className="sidebar-right">
                        {this.closeIcon()}
                        <div className="grid-block shrink vertical">
                            <ItemProvenance item={{name, value, provenance}}/>
                        </div>
                    </div>
                </Sidebar>
            </SidebarPortal> : null;
    }

    hasData() {
        const {name, value, provenance} = this.props;
        return !!name && name.length > 0
            && !!value && name.length > 0
            && !!provenance && typeof provenance === "object";
    }

    closeIcon() {
        return <a className="float-right close-icon" onClick={this.onClose}>
            <i className="fa fa-close"/>
        </a>;
    }

    onClose() {
        const {actions} = this.props;
        actions.closeItemDetails();
    }
}

const mapStateToProps = function (state, ownProps) {
    const {itemProvenance} = state;
    const {studyId} = ownProps;
    return {
        studyId,
        ...itemProvenance
    }
};

export const mapDispatchToProps = function (dispatch) {
    return {
        actions: {
            fetchItemProvenance: (payload) => dispatch(fetchItemProvenance(payload)),
            fetchSubmittedItemProvenance: (payload) => dispatch(fetchSubmittedItemProvenance(payload)),
            closeItemDetails: () => dispatch(closeItemDetails()),
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(ProvenanceSidebar);