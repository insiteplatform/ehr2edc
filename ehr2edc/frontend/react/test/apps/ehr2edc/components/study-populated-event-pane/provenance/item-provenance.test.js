import React from "react"
import {ItemProvenance} from "../../../../../../src/apps/ehr2edc/components/study-populated-event-pane/provenance/item-provenance";
import readDataFromFile from "../../../../../__helper__/sample-data-file";

describe("Item Provenance", () => {
    test("Component does not render without item", () => {
        // Given: A ItemProvenance component
        const comp = mount(<ItemProvenance/>);

        // When: No item is present
        comp.setProps({item: undefined});
        comp.update();

        // Then: No component is rendered
        expect(comp).toEqual({});
    });

    test("Component renders loading state", () => {
        // Given: A ItemProvenance component
        const comp = mount(<ItemProvenance/>);

        // When: An item is present
        comp.setProps({
            item: {
                name: "Lab Test or Examination Short Name",
                value: "VAL",
                provenance: undefined
            }
        });

        // Then: The component is rendered
        expect(comp).toMatchSnapshot();
    });

    test("Component renders error state", () => {
        // Given: A ItemProvenance component
        const comp = mount(<ItemProvenance/>);

        // When: An item is present
        comp.setProps({
            item: {
                name: "Lab Test or Examination Short Name",
                value: "VAL",
                provenance: {
                    error: {
                        response: {
                            data: {
                                "issues": [{
                                    "reference": null,
                                    "field": null,
                                    "message": "Something happened in the back-end"
                                }]
                            }
                        }
                    },
                    data: undefined
                }
            }
        });

        // Then: The component is rendered
        expect(comp).toMatchSnapshot();
    });

    test("Component renders success state", () => {
        // Given: A ItemProvenance component
        const comp = mount(<ItemProvenance/>);

        // When: An item is present
        comp.setProps({
            item: {
                name: "Lab Test or Examination Short Name",
                value: "VAL",
                provenance: {
                    error: undefined,
                    data: {
                        "groups": [
                            {
                                "label": "Group 1",
                                "items": [
                                    {
                                        "label": "Item 1",
                                        "value": "Value 1"
                                    },
                                    {
                                        "label": "Item 2",
                                        "value": "Value 2"
                                    },
                                    {
                                        "label": "Item 3",
                                        "value": ""
                                    },
                                    {
                                        "label": "Item 4",
                                        "value": undefined
                                    }
                                ]
                            }
                        ],
                        "items": [
                            {
                                "label": "Item 1",
                                "value": "Value 1"
                            },
                            {
                                "label": "Item 2",
                                "value": "Value 2"
                            },
                            {
                                "label": "Item 3",
                                "value": ""
                            },
                            {
                                "label": "Item 4",
                                "value": undefined
                            }
                        ]
                    }
                }
            }
        });

        // Then: The component is rendered
        expect(comp).toMatchSnapshot();
    })
});

describe("Provenance listing", () => {

    test("Items are rendered", () => {
        // Given: A provenance object containing items
        const provenance = {
            groups: [],
            items: [
                {label: "Item 1", value: "Value 1"},
                {label: "Item 2", value: "Value 2"}
            ]
        };

        // When: Generating the listing
        const listing = ProvenanceListing.for(provenance);

        // Then: All items are rendered correctly
        listing.hasLength(2);
        listing.hasListingFor("Item 1", "Value 1");
        listing.hasListingFor("Item 2", "Value 2");
    });

    test("Items within a group are rendered as sublisting of that group", () => {
        // Given: A provenance object containing another object
        const provenance = {
            groups: [{
                "label": "Group 1",
                "items": [
                    {
                        "label": "Item 1",
                        "value": "Value 1"
                    },
                    {
                        "label": "Item 2",
                        "value": "Value 2"
                    }
                ]
            }],
            items: []
        };

        // When: Generating the listing
        const listing = ProvenanceListing.for(provenance);

        // Then: A group is created
        listing.containsListing("Group 1");
        const objectListing = listing.sublistingFor("Group 1");
        // And: The items in the group are rendered correctly
        objectListing.hasLength(2);
        objectListing.hasListingFor("Item 1", "Value 1");
        objectListing.hasListingFor("Item 2", "Value 2");
    });

    test("Empty provenance gets rendered correctly", () => {
        const provenance = {
            groups: [],
            items: []
        };

        const listing = ProvenanceListing.for(provenance);

        listing.hasLength(0);
    });
});

describe("Application REST API complies with the component", () => {
    test("The item provenance REST API complies", () => {
        // Given: An item provenance REST response
        const response = readDataFromFile("eventcontroller/getItemProvenance.json");

        // When: Generating the listing
        const listing = ProvenanceListing.for(response);

        // Then: The provenance is rendered correctly
        assertListing(listing)
    });

    test("The submitted item provenance REST API complies", () => {
        // Given: A submitted item provenance REST response
        const response = readDataFromFile("reviewedeventcontroller/getSubmittedItemProvenance.json");

        // When: Generating the listing
        const listing = ProvenanceListing.for(response);

        // Then: The provenance is rendered correctly
        assertListing(listing)
    });

    function assertListing(listing) {
        listing.hasLength(7);
        listing.hasListingFor("Item 4", "Value 4");
        listing.hasListingFor("Item 5", "none");
        listing.hasListingFor("Item 6", "none");
        const group = listing.sublistingFor("Group 1");
        group.hasLength(3);
        group.hasListingFor("Item 1", "Value 1");
        group.hasListingFor("Item 2", "none");
        group.hasListingFor("Item 3", "none");
    }
});

const ProvenanceListing = function () {
    const _render = function (provenance) {
        return shallow(ItemProvenance.prototype.provenanceDataListing(provenance));
    };
    const _Assertions = function (listing) {
        const _allListings = function () {
            return listing.find(".row");
        };
        const _findGroup = function (key) {
            return _allListings().filterWhere(row => row.children("h6").exists() && row.children("h6").text() === key)
        };
        const _findListing = function (key) {
            return _allListings().filterWhere(row => row.children("dt").exists() && row.children("dt").text() === key)
        };
        const _listingValueMatches = function (key, value) {
            const listingValue = _findListing(key).children("dd");
            if (value.length === 1) {
                expect(listingValue.text() === value[0]).toBe(true);
            } else {
                expect(
                    value.map(val => listingValue.find("li").map(elem => elem.text()).includes(val)).includes(false)
                ).toBe(false)
            }
        };

        return {
            hasLength: function (length) {
                expect(_allListings()).toHaveLength(length);
            },
            containsListing: function (key) {
                const ungrouped = _findListing(key).exists();
                const group = _findGroup(key).exists();
                expect(ungrouped || group).toBe(true);
            },
            hasListingFor: function (key, ...value) {
                this.containsListing(key);
                _listingValueMatches(key, value);
            },
            sublistingFor: function (key) {
                const sublisting = _findGroup(key).children("dl");
                return _Assertions(sublisting);
            }
        }
    };

    return {
        for: function (provenance) {
            const listing = _render(provenance);
            return _Assertions(listing);
        }
    }
}();