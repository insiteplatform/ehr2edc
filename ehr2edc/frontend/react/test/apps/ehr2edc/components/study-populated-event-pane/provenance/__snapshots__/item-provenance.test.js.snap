// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Provenance Component renders error state 1`] = `
<ItemProvenance
  item={
    Object {
      "name": "Lab Test or Examination Short Name",
      "provenance": Object {
        "data": undefined,
        "error": Object {
          "response": Object {
            "data": Object {
              "issues": Array [
                Object {
                  "field": null,
                  "message": "Something happened in the back-end",
                  "reference": null,
                },
              ],
            },
          },
        },
      },
      "value": "VAL",
    }
  }
>
  <h4
    className="facts-title"
  >
    Source Data Details
  </h4>
  <dl
    className="margin-none"
  >
    <h5>
      Name
    </h5>
    <dd>
      Lab Test or Examination Short Name
    </dd>
    <h5>
      Submitted Value
    </h5>
    <dd>
      VAL
    </dd>
  </dl>
  <h5>
    Source Data
  </h5>
  <div
    className="alert-box alert warning margin-trl"
  >
    <RequestError
      error={
        Object {
          "response": Object {
            "data": Object {
              "issues": Array [
                Object {
                  "field": null,
                  "message": "Something happened in the back-end",
                  "reference": null,
                },
              ],
            },
          },
        }
      }
    >
      <Error
        error={
          Object {
            "issues": Array [
              Object {
                "field": null,
                "message": "Something happened in the back-end",
                "reference": null,
              },
            ],
          }
        }
      >
        <Issues
          issues={
            Array [
              Object {
                "field": null,
                "message": "Something happened in the back-end",
                "reference": null,
              },
            ]
          }
        >
          <ul
            className="bullet"
          >
            <Issue
              issue={
                Object {
                  "field": null,
                  "message": "Something happened in the back-end",
                  "reference": null,
                }
              }
              key="0"
            >
              <ErrorMessage
                message="Something happened in the back-end"
              >
                <li>
                  Something happened in the back-end
                </li>
              </ErrorMessage>
            </Issue>
          </ul>
        </Issues>
      </Error>
    </RequestError>
  </div>
</ItemProvenance>
`;

exports[`Item Provenance Component renders loading state 1`] = `
<ItemProvenance
  item={
    Object {
      "name": "Lab Test or Examination Short Name",
      "provenance": undefined,
      "value": "VAL",
    }
  }
>
  <h4
    className="facts-title"
  >
    Source Data Details
  </h4>
  <dl
    className="margin-none"
  >
    <h5>
      Name
    </h5>
    <dd>
      Lab Test or Examination Short Name
    </dd>
    <h5>
      Submitted Value
    </h5>
    <dd>
      VAL
    </dd>
  </dl>
</ItemProvenance>
`;

exports[`Item Provenance Component renders success state 1`] = `
<ItemProvenance
  item={
    Object {
      "name": "Lab Test or Examination Short Name",
      "provenance": Object {
        "data": Object {
          "groups": Array [
            Object {
              "items": Array [
                Object {
                  "label": "Item 1",
                  "value": "Value 1",
                },
                Object {
                  "label": "Item 2",
                  "value": "Value 2",
                },
                Object {
                  "label": "Item 3",
                  "value": "",
                },
                Object {
                  "label": "Item 4",
                  "value": undefined,
                },
              ],
              "label": "Group 1",
            },
          ],
          "items": Array [
            Object {
              "label": "Item 1",
              "value": "Value 1",
            },
            Object {
              "label": "Item 2",
              "value": "Value 2",
            },
            Object {
              "label": "Item 3",
              "value": "",
            },
            Object {
              "label": "Item 4",
              "value": undefined,
            },
          ],
        },
        "error": undefined,
      },
      "value": "VAL",
    }
  }
>
  <h4
    className="facts-title"
  >
    Source Data Details
  </h4>
  <dl
    className="margin-none"
  >
    <h5>
      Name
    </h5>
    <dd>
      Lab Test or Examination Short Name
    </dd>
    <h5>
      Submitted Value
    </h5>
    <dd>
      VAL
    </dd>
  </dl>
  <h5>
    Source Data
  </h5>
  <div>
    <div
      className="row"
      key="0"
    >
      <h6>
        Group 1
      </h6>
      <dl
        className="responsive"
      >
        <div
          className="row"
          key="0"
        >
          <dt>
            Item 1
          </dt>
          <dd>
            Value 1
          </dd>
        </div>
        <div
          className="row"
          key="1"
        >
          <dt>
            Item 2
          </dt>
          <dd>
            Value 2
          </dd>
        </div>
        <div
          className="row"
          key="2"
        >
          <dt>
            Item 3
          </dt>
          <dd>
            <i
              className="disabled"
            >
              none
            </i>
          </dd>
        </div>
        <div
          className="row"
          key="3"
        >
          <dt>
            Item 4
          </dt>
          <dd>
            <i
              className="disabled"
            >
              none
            </i>
          </dd>
        </div>
      </dl>
    </div>
    <dl
      className="responsive"
    >
      <div
        className="row"
        key="0"
      >
        <dt>
          Item 1
        </dt>
        <dd>
          Value 1
        </dd>
      </div>
      <div
        className="row"
        key="1"
      >
        <dt>
          Item 2
        </dt>
        <dd>
          Value 2
        </dd>
      </div>
      <div
        className="row"
        key="2"
      >
        <dt>
          Item 3
        </dt>
        <dd>
          <i
            className="disabled"
          >
            none
          </i>
        </dd>
      </div>
      <div
        className="row"
        key="3"
      >
        <dt>
          Item 4
        </dt>
        <dd>
          <i
            className="disabled"
          >
            none
          </i>
        </dd>
      </div>
    </dl>
  </div>
</ItemProvenance>
`;
